### PR TITLE
fix(batch): better error handling for single value

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Report an error instead of panic if `single_value` occurs more than one row.
- Better error display.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#5114

### CODE REVIEW

File: src/batch/src/executor/hash_agg.rs [hunk 1]
```diff
@@ -207,9 +207,9 @@ impl<K: HashKey + Send + Sync> HashAggExecutor<K> {
                 });
 
                 // TODO: currently not a vectorized implementation
-                states
-                    .iter_mut()
-                    .for_each(|state| state.update_single(&chunk, row_id).unwrap());
```
Comment (BugenZhao) [inline src/batch/src/executor/hash_agg.rs:212]: Oops. I believe there're more cases of unwrapping the result in closure in our system, just for no idea of how to throw the error. 😢

**Review by BugenZhao — APPROVED** [review]
LGTM

File: src/frontend/test_runner/tests/testdata/range_scan.yaml [hunk 2]
```diff
@@ -32,22 +32,22 @@
...
   sql: |
     SELECT * FROM orders_count_by_user WHERE user_id = 2147483647 + 1
-  batch_error: 'Expr error: NumericOutOfRange'
+  batch_error: 'Expr error: Out of range'
 - before:
   - create_table_and_mv
   sql: |
     SELECT * FROM orders_count_by_user WHERE user_id = 'a'
-  batch_error: 'Expr error: Cast("str", "i64")'
+  batch_error: 'Expr error: Can''t cast str to i64'
```
Comment (xxchan) [inline src/frontend/test_runner/tests/testdata/range_scan.yaml:45]: What's wrong 😄

Comment (chenzl25) [reply src/frontend/test_runner/tests/testdata/range_scan.yaml:45]: Previously errors were displayed in debug mode.

File: src/batch/src/executor/hash_agg.rs (hunk 1 above)
Comment (chenzl25) [reply src/batch/src/executor/hash_agg.rs:212]: Yes.😢

File: src/frontend/test_runner/tests/testdata/range_scan.yaml (hunk 2 above)
Comment (xxchan) [reply src/frontend/test_runner/tests/testdata/range_scan.yaml:45]: I mean the `Can''t` has an extra `'`...

Comment (BugenZhao) [reply src/frontend/test_runner/tests/testdata/range_scan.yaml:45]: Escape char of yaml?

Comment (chenzl25) [reply src/frontend/test_runner/tests/testdata/range_scan.yaml:45]: Yes, it is an escape char, because of `'` `'` around `Can't`

Comment (xxchan) [reply src/frontend/test_runner/tests/testdata/range_scan.yaml:45]: got it

**Pull request merged**